File: ToolBag/ParseTool/src/parsetool/workflow/models/expressions/Conditional_Expression.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package parsetool.workflow.models.expressions;

import parsetool.models.common.TokenPair;

/**
 *
 * @author cqy
 */
public class Conditional_Expression extends TokenPair{
    private Logical_Or_Expression logical_or_expression;
    private Expression second_expression;
    private Conditional_Expression third_expression;

    /**
     * @return the logical_or_expression
     */
    public Logical_Or_Expression getLogical_or_expression() {
        return logical_or_expression;
    }

    /**
     * @param logical_or_expression the logical_or_expression to set
     */
    public void setLogical_or_expression(Logical_Or_Expression logical_or_expression) {
        this.logical_or_expression = logical_or_expression;
    }

    /**
     * @return the second_expression
     */
    public Expression getSecond_expression() {
        return second_expression;
    }

    /**
     * @param second_expression the second_expression to set
     */
    public void setSecond_expression(Expression second_expression) {
        this.second_expression = second_expression;
    }

    /**
     * @return the third_expression
     */
    public Conditional_Expression getThird_expression() {
        return third_expression;
    }

    /**
     * @param third_expression the third_expression to set
     */
    public void setThird_expression(Conditional_Expression third_expression) {
        this.third_expression = third_expression;
    }
}

File: CodeHelper/CodeHelper.Core/Parse/ParseResults/Workflows/Expressions/Conditional_Expression.cs
using System;
using System.Collections.Generic;
namespace CodeHelper.Core.Parse.ParseResults.Workflows.Expressions
{
    public class Conditional_Expression : TokenPair
    {
        public Logical_Or_Expression Logical_or_expression { get; set; }
        public Expression Second_expression { get; set; }
        public Conditional_Expression Third_expression { get; set; }

        public ConditionalExpressionInfo Parse()
        {
            var rslt = new ConditionalExpressionInfo(this);
            rslt.Logical_Or_Expression = this.Logical_or_expression.Parse();
            if (Second_expression != null)
            {
                rslt.Expression = this.Second_expression.Parse();
            }
            if (Third_expression != null)
            {
                rslt.ConditionalExpression = this.Third_expression.Parse();
            }
            return rslt;
        }
    }
}